src/code/z_kankyo.c

```diff
@@ -1668,16 +1777,16 @@ void Environment_DrawRain(GlobalContext* globalCtx, View* view, GraphicsContext*
     }
 }
 
-void func_80074CE8(GlobalContext* globalCtx, u32 arg1) {
-    if ((globalCtx->envCtx.unk_BD != arg1) && (globalCtx->envCtx.unk_D8 >= 1.0f) &&
-        (globalCtx->envCtx.unk_BF == 0xFF)) {
-        if (arg1 > 30) {
-            arg1 = 0;
+void Environment_ChangeLightSetting(GlobalContext* globalCtx, u32 lightSetting) {
+    if ((globalCtx->envCtx.lightSetting != lightSetting) && (globalCtx->envCtx.lightBlend >= 1.0f) &&
+        (globalCtx->envCtx.lightSettingOverride == LIGHT_SETTING_OVERRIDE_NONE)) {
+        if (lightSetting > LIGHT_SETTING_MAX) {
```

Comment:
I feel like this should be
```suggestion
        if (lightSetting >= LIGHT_SETTING_MAX) {
```
with `LIGHT_SETTING_MAX` being 31

it would also ditch the +1 in `lightSettingOverride <= LIGHT_SETTING_MAX + 1`

I'm not sure what is this `LIGHT_SETTING_MAX` value about though, `sTimeBasedLightConfigs` references 0 to 15 and 20 to 23 boundaries included, would be 24 settings, and 7 bonus settings (24-30)? looks kinda random
(`ganontika_sceneLightSettings0x000300` has 26 settings, most in the game afaict)

If anything this should use `envCtx->numLightSettings`
I also don't see why the check in `Environment_Update` would allow 31 but this doesn't

I know you can't fix the jank code but I'm questioning if the define makes sense to have. Maybe it does and the check in `Environment_Update` is a typo that should be `<` instead of `<=`, or they just used two different max values that happened to be 1 off